guides-markdown/connect-local-nodes.md

```diff
@@ -0,0 +1,95 @@
+# Connect two local nodes
+
+```post-author
+Orfeas Stefanos Thyfronitis Litos
+```
+
+```post-description
+Start and manually connect two regtest nodes in a single script.
+```
+
+When we start two nodes in the regtest (loopback network, 127.0.0.1) with
+manually set ports, they don't get connected automatically because they don't
+know each other's port. We thus have to connect them manually. In this example
+we will fire up one SPV and one Full node and initiate a connection from the
+SPV side. The SPV node is given the address of the Full node and attempts to
+connect. If everything goes well, the two nodes connect. Finally they are
+gracefully closed.
+
+The script should work out of the box, the only requirement is having `bcoin`
+installed.
+
+```javascript
+const bcoin = require('bcoin').set('regtest');
+const NetAddress = bcoin.net.NetAddress;
+const Network = bcoin.Network;
+const assert = require('assert');
+
+async function delay(ms) {
+  return new Promise(resolve => {
+    setTimeout(resolve, ms);
+  });
+}
+
+const regtest = Network.get().toString();
+
+// create nodes
+const spvNode = new bcoin.SPVNode({
+  network: regtest,
+  httpPort: 48449 // avoid clash of ports
+});
+
+const fullNode = new bcoin.FullNode({
+  network: regtest,
+  port: 48445,
+  bip37: true, // accept SPV nodes
+  listen: true
+});
+// nodes created!
+
+(async () => {
+  // start nodes
+  await spvNode.open();
+  await fullNode.open();
+
+  await spvNode.connect();
+  await fullNode.connect();
+  // nodes started!
+
+  // get peer from known address
+  const addr = new NetAddress({
+    host: '127.0.0.1',
+    port: fullNode.pool.options.port
+  });
+  const peer = spvNode.pool.createOutbound(addr);
+
+  // allow some time for spvNode to figure
+  // out that its peer list is empty
+  await delay(800);
+
+  // no peers for the spvNode yet :(
+  console.log('spvNode\'s peers before connection:', spvNode.pool.peers.head());
+  assert.equal(spvNode.pool.peers.head(), null);
+
+  // connect spvNode with fullNode
+  spvNode.pool.peers.add(peer);
+
+  // allow some time to establish connection
+  await delay(3000);
```

Comment:
I actually had to change this delay to 10 seconds `delay(10000)` on my machine or the second assert would fail. I wonder if we could use an await/async method here instead of gambling on how long the connection will take? I haven't tested this yet but maybe `peer.open()`? https://github.com/bcoin-org/bcoin/blob/master/lib/net/peer.js#L347-L361